workflow/executor/executor.go

```diff
@@ -806,7 +806,20 @@ func (we *WorkflowExecutor) AddError(err error) {
 
 // AddAnnotation adds an annotation to the workflow pod
 func (we *WorkflowExecutor) AddAnnotation(key, value string) error {
-	return common.AddPodAnnotation(we.ClientSet, we.PodName, we.Namespace, key, value)
+	var err error
+	_ = wait.ExponentialBackoff(retry.DefaultRetry, func() (bool, error) {
```

Comment:
Shouldn't we get the `err` from the return value of `wait.ExponentialBackoff` instead of just assigning it inside the anonymous function? I think using the return value is more idiomatic and less bug-prone.